joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/json/ClassTests.scala

```diff
@@ -0,0 +1,1018 @@
+          }
+
+          inside(bodyMethod.block.astChildren.l) {
+            case (one: Literal) :: Nil =>
+              one.code shouldBe "1"
+              one.typeFullName shouldBe s"${GlobalTypes.kernelPrefix}.Integer"
+            case xs => fail(s"Expected one literal, got [${xs.code.mkString(",")}]")
+          }
+
+        case xs => fail(s"Expected body method and init method, got [${xs.code.mkString(",")}]")
+      }
+    }
+  }
+
+  "Class defined in Namespace" ignore {
+    val cpg = code("""
+        |class Api::V1::MobileController
+        |end
+        |""".stripMargin)
+
+    inside(cpg.namespaceBlock.fullNameExact("Api.V1").typeDecl.l) {
+      case mobileNamespace :: mobileClassNamespace :: Nil =>
+        mobileNamespace.name shouldBe "MobileController"
+        mobileNamespace.fullName shouldBe "Test0.rb:<main>.Api.V1.MobileController"
+
+        mobileClassNamespace.name shouldBe "MobileController<class>"
+        mobileClassNamespace.fullName shouldBe "Test0.rb:<main>.Api.V1.MobileController<class>"
+      case xs => fail(s"Expected two namespace blocks, got ${xs.code.mkString(",")}")
+    }
+
+    inside(cpg.typeDecl.name("MobileController").l) {
+      case mobileTypeDecl :: Nil =>
+        mobileTypeDecl.name shouldBe "MobileController"
+        mobileTypeDecl.fullName shouldBe "Test0.rb:<main>.Api.V1.MobileController"
+        mobileTypeDecl.astParentFullName shouldBe "Api.V1"
+        mobileTypeDecl.astParentType shouldBe NodeTypes.NAMESPACE_BLOCK
+
+        mobileTypeDecl.astParent.isNamespaceBlock shouldBe true
+
+        val namespaceDecl = mobileTypeDecl.astParent.asInstanceOf[NamespaceBlock]
+        namespaceDecl.name shouldBe "Api.V1"
+        namespaceDecl.filename shouldBe "Test0.rb"
+
+        namespaceDecl.astParent.isFile shouldBe true
+        val parentFileDecl = namespaceDecl.astParent.asInstanceOf[File]
+        parentFileDecl.name shouldBe "Test0.rb"
+
+      case xs => fail(s"Expected one class decl, got [${xs.code.mkString(",")}]")
+    }
+  }
+
+  "Namespace scope is popping properly" in {
+    val cpg = code("""
+        |class Foo::Bar
+        |end
+        |
+        |class Baz
+        |end
+        |""".stripMargin)
+
+    inside(cpg.typeDecl.name("Baz").l) {
+      case bazTypeDecl :: Nil =>
+        bazTypeDecl.fullName shouldBe "Test0.rb:<main>.Baz"
+      case xs => fail(s"Expected one type decl, got [${xs.code.mkString(",")}]")
+    }
+  }
+
+  "Self param in static method" in {
+    val cpg = code("""
+        |class Benefits < ApplicationRecord
+        |def self.save(file, backup = false)
+        |    data_path = Rails.root.join("public", "data")
+        |    full_file_name = "#{data_path}/#{file.original_filename}"
+        |    f = File.open(full_file_name, "wb+")
+        |    f.write file.read
+        |    f.close
+        |    make_backup(file, data_path, full_file_name) if backup == "true"
+        |end
+        |end
+        |""".stripMargin)
```

Comment:
Is there a reason we aren't testing for anything here?

Reply:
No this must be a bug carried through from the original classes, but a good catch